helix-term/src/commands.rs

```diff
@@ -3820,12 +3850,21 @@ pub fn completion(cx: &mut Context) {
     let start_offset = cursor.saturating_sub(offset);
     let prefix = text.slice(start_offset..cursor).to_string();
 
+    doc.savepoint();
+    let trigger_version = doc.version();
     cx.callback(
         future,
         move |editor, compositor, response: Option<lsp::CompletionResponse>| {
-            let doc = doc!(editor);
+            let doc = doc_mut!(editor);
+            let savepoint = match doc.savepoint.take() {
+                Some(s) => s,
+                None => return,
```

Comment:
In what case this will happen? Can you please leave a comment?

Reply:
It's getted result after the mode was changed to normal and was changed to insert again.